File: Assets/AltTester/Runtime/AltDriver/Commands/ObjectCommands/AltGetVisualElementProperty.cs
/*
    Copyright(C) 2026 Altom Consulting
*/

namespace AltTester.AltTesterSDK.Driver.Commands
{
    public class AltGetVisualElementProperty<T> : AltBaseCommand
    {
        AltGetVisualElementPropertyParams cmdParams;
        public AltGetVisualElementProperty(IDriverCommunication commHandler, string propertyName, AltObject altObject) : base(commHandler)
        {
            cmdParams = new AltGetVisualElementPropertyParams(altObject, propertyName);
        }
        public T Execute()
        {
            CommHandler.Send(cmdParams);
            T result = CommHandler.Recvall<T>(cmdParams);

            // If the result is an AltObject, ensure its CommHandler is set
            if (result is AltObject altObject)
            {
                altObject.CommHandler = CommHandler;
            }

            return result;
        }
    }
}

File: Bindings~/java/src/main/java/com/alttester/Commands/UnityCommand/AltGetVisualElementProperty.java
/*
    Copyright(C) 2026 Altom Consulting
*/

package com.alttester.Commands.UnityCommand;

import com.alttester.Commands.AltBaseCommand;
import com.alttester.IMessageHandler;

public class AltGetVisualElementProperty extends AltBaseCommand {

  private AltGetVisualElementProperyParams params;

  public AltGetVisualElementProperty(
      IMessageHandler messageHandler,
      AltGetVisualElementProperyParams altGetVisualElementProperyParams) {
    super(messageHandler);
    this.params = altGetVisualElementProperyParams;
    this.params.setCommandName("getVisualElementProperty");
  }

  public <T> T Execute(Class<T> returnType) {

    SendCommand(params);
    return recvall(params, returnType);
  }
}
